### PR TITLE
Added getPosition() and getSize() to sf::Rect<T>.

### PR DESCRIPTION
**Pull request by TankOs**
Suggestion to add `getPosition()` and `getSize()` to `sf::Rect`.

I'm aware that it can be easily done by the user as well, but to be honest, I miss this all the time when I create a new SFML project. Doing calculations is easier and better understandable.

I'm just not sure if the user should be warned of the copies the functions create.

### CODE REVIEW

**Comment by mantognini** [comment]
[There was a similar feature request a while back](http://en.sfml-dev.org/forums/index.php?topic=16218.0). Maybe we should restart the discussion from there if new arguments are made, shouldn't we?

**Comment by TankOs** [comment]
http://en.sfml-dev.org/forums/index.php?topic=16218.msg141828#msg141828

File: include/SFML/Graphics/Rect.hpp [hunk 1]
```diff
@@ -146,6 +146,26 @@ class Rect
     bool intersects(const Rect<T>& rectangle, Rect<T>& intersection) const;
 
     ////////////////////////////////////////////////////////////
+    /// \brief Get the position of the rectangle
+    ///
+    /// \return Position of rectangle
```
Comment (binary1248) [inline include/SFML/Graphics/Rect.hpp]: I would explicitly mention that this is the position of the top-left corner of the rectangle, just like how it is done in the constructor.

Comment (TankOs) [inline include/SFML/Graphics/Rect.hpp]: Done

**Comment by TankOs** [comment]
Bump

**Comment by mantognini** [comment]
I personally don't know about this: yes, as you said, this is handy, but is this enough to add it? I don't know so I raise the white flag.

But regarding this:
> I'm just not sure if the user should be warned of the copies the functions create.

There're way much heavier tasks that the user has to care about. Those copies will most probably be elided by the compiler, anyway.

**Comment by Bromeon** [comment]
I posted on the forum, let's discuss the need and general design (and bump) there :wink:

**Comment by Bromeon** [comment]
In the past, I often needed to align drawables to coordinate bounds, Code of the kind `a.setPosition(a.getPosition().x, b.getGlobalBounds().y)` is very annoying to maintain.

I was thinking of convenience helpers such as
```
float x = sf::Layout::getLeft(sprite);
float w = sf::Layout::getWidth(shape);
sf::Vector2f pos = sf::Layout::getRightBottom(text);

sf::Layout::setRight(sprite2, x + w);
```

Such helpers could be overloaded for all "alignable" objects, including `sf::FloatRect`. This would not just reduce a lot of boilerplate that is currently needed, but also provide a uniform API across different types. Setters could easily be provided, too.

**Pull request merged**